File: be-api/src/main/java/cd/beapi/dto/response/AppointmentResponse.java
package cd.beapi.dto.response;

import cd.beapi.enumerate.AppointmentStatus;

import java.time.Instant;
import java.time.LocalDateTime;

public record AppointmentResponse(
        Long id,
        String code,
        LocalDateTime appointmentDate,
        Instant createdAt,
        Instant modifiedAt,
        String symptom,
        String note,
        AppointmentStatus status,
        Integer queueNumber,
        Long version,
        Long patientId,
        String patientCode,
        String patientName,
        String patientPhone,
        Long dentistId,
        String dentistCode,
        String dentistName,
        Long receptionistId,
        String receptionistName,
        String snapshotPatientName,
        String snapshotPatientPhone
) {}

File: fe-client/src/app/core/model/response/appointment-response.ts
export type AppointmentStatus =
  | 'PENDING'
  | 'CONFIRMED'
  | 'IN_PROGRESS'
  | 'COMPLETED'
  | 'CANCELLED';

export type AppointmentArrivalStatus = 'NOT_ARRIVED' | 'ARRIVED' | 'NO_SHOW';

export interface AppointmentResponse {
  id: number;
  code: string;
  appointmentDate: string;
  createdAt: string;
  modifiedAt: string | null;
  estimatedDurationMinutes: number;
  symptom: string | null;
  note: string | null;
  status: AppointmentStatus;
  arrivalStatus: AppointmentArrivalStatus | null;
  queueNumber: number | null;
  version: number;
  patientId: number;
  patientCode: string;
  dentistName: string | null;
  patientName: string;
  patientPhone: string | null;
}
